src/Gelf4Net.Core/Util/BufferedLogSender.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Gelf4Net.Util
+{
+    public class BufferedLogSender
+    {
+        private readonly BlockingCollection<string> _pendingTasks;
+        private readonly CancellationTokenSource _cts;
+        private readonly Func<string, Task> _sendFunc;
+        private readonly Task _sender;
+
+        public BufferedLogSender(Func<string, Task> sendFunc)
+        {
+            _pendingTasks = new BlockingCollection<string>(100);
```

Comment:
Is a blocking collection the best thing to use for this? Shouldn't we be using a structure where we don't actually have to block the running thread when adding or removing messages?

Reply:
Log4net's append method isn't async, so we are going to end up blocking that anyway. If we are blocking the background thread that's sending logging to the HTTP target, that's less of an issue as you'd expect no logging to mean that the process is not up to much that you'd be contending with.

Reply:
I am more concerned with blocking when the process is busy and the BlockingCollection is full causing all 4 sending threads from the threadpool to block when they could be doing something else.